bin/tx-prover/README.md

```diff
@@ -1,84 +1,89 @@
-```
-
-Both services can be installed at once from the source code using specific git revisions with `cargo install`. Note that since these aren't official releases we cannot provide much support for any issues you run into, so consider this for advanced users only.
+cargo install miden-tx-prover --locked --version x.y.z
+``` -->
 
-If you want to build from a local version, from the root of the workspace you can run:
+To build the CLI from a local version, from the root of the workspace you can run:
 
 ```bash
-make install-prover-worker
-make install-prover-proxy
+make install-tx-prover
 ```
 
+The CLI can be installed from the source code using specific git revisions with `cargo install`. Note that since these aren't official releases we cannot provide much support for any issues you run into, so consider this for advanced users only.
+
 Note that for the prover worker you might need to enable the `testing` feature in case the transactions were executed with reduced proof-of-work requirements (or otherwise, the proving process will fail). This step will also generate the necessary protobuf-related files. You can achieve that by generating the binary with:
 
 ```bash
-make install-prover-worker-testing
+make install-tx-prover-testing
 ```
 
-## Running the Service
+## Usage
 
-Once installed, you can run the worker with:
+Once installed, you need to initialize the CLI with:
 
 ```bash
-RUST_LOG=info miden-tx-prover-worker
+miden-tx-prover init
 ```
 
-By default, the server will start on `0.0.0.0:50051`. You can change this and the log level by setting the following environment variables:
+This will create the `miden-tx-prover.toml` file in your current directory. This file will hold the configuration for the proxy. You can modify the configuration by changing the host and ports of the services, and add workers. An example of a valid configuration is:
 
-```bash
-PROVER_SERVICE_HOST=<your-host>
-PROVER_SERVICE_PORT=<your-port>
-RUST_LOG=<log-level>
-```
+```toml
+host = "0.0.0.0"
+port = 8082
+timeout_secs = 100
+connection_timeout_secs = 10
+max_queue_items = 10
+max_retries_per_request = 1
+max_req_per_sec = 5
 
-And to run the proxy:
+[[workers]]
+host = "0.0.0.0"
+port = 8083
 
-```bash
-RUST_LOG=info miden-tx-prover-proxy
+[[workers]]
+host = "0.0.0.0"
+port = 8084
 ```
 
-By default, the server will start on `0.0.0.0:6188`. This can be changed by setting:
+To add more workers, you will need to add more items with the `[[workers]]` tags.
+
+To start the worker service you will need to run:
 
 ```bash
-PROXY_HOST=<your-host>
-PROXY_PORT=<your-port>
+miden-tx-prover start-worker --host 0.0.0.0 --port 8082
 ```
 
-Also, it is mandatory to set at least one prover worker by setting the `PROVER_WORKERS` env var:
+This will a worker using the hosts and ports defined in the command options. In case that one of the values is not present, the CLI will default to `0.0.0.0` for the host and `50051` for the port.
```

Comment:
```suggestion
This will spawn a worker using the hosts and ports defined in the command options. In case that one of the values is not present, the CLI will default to `0.0.0.0` for the host and `50051` for the port.
```